src/target/riscv/riscv.c

```diff
@@ -1601,9 +1654,21 @@ static int riscv_trigger_detect_hit_bits(struct target *target, int64_t *unique_
 				break;
 			case CSR_TDATA1_TYPE_MCONTROL:
 				hit_mask = CSR_MCONTROL_HIT;
+				r->need_single_step = true;
```

Comment:
```suggestion
				*need_single_step = true;
```

Please, add also these two lines to the top of the function. The variable `need_single_step` should always be assigned a valid value. For the sake of code clarity, we should not require the caller to initialize it to `false`:

```c
assert(need_single_step);
*need_single_step = false;
```

Reply:
addressed